src/main/java/io/swagger/swaggerhub/plugin/SwaggerHubClient.java

```diff
@@ -26,24 +31,60 @@
     private Log log;
     private String basePath;
 
-    public SwaggerHubClient(String host, int port, String protocol, String token, Log log, String basePath) {
-        client = setupHttpClient();
+    public SwaggerHubClient(String host, int port, String protocol, String token, Log log, String basePath, org.apache.maven.settings.Proxy proxy) {
+        client = setupHttpClient(proxy);
         this.host = host;
         this.port = port;
         this.protocol = protocol;
         this.token = token;
-        this.log=log;
+        this.log = log;
         this.basePath = basePath;
     }
 
-    private OkHttpClient setupHttpClient(){
-        OkHttpClient client = new OkHttpClient();
-        client.setConnectTimeout(30, TimeUnit.SECONDS);
-        client.setReadTimeout(30, TimeUnit.SECONDS);
-        client.setWriteTimeout(30, TimeUnit.SECONDS);
+    private OkHttpClient setupHttpClient(org.apache.maven.settings.Proxy proxy) {
+
+        OkHttpClient.Builder okHttpClientBuilder = new OkHttpClient.Builder();
+        if (proxy != null) {
+            Proxy.Type proxyType = getProxyType(proxy.getProtocol());
+            Proxy httpProxy = new Proxy(proxyType,
+                    new InetSocketAddress(proxy.getHost(), proxy.getPort()));
+            okHttpClientBuilder.proxy(httpProxy);
+            if (StringUtils.isNotEmpty(proxy.getUsername())) {
+                Authenticator proxyAuthenticator = new Authenticator() {
+                    @Override
+                    public Request authenticate(Route route, Response response) throws IOException {
+                        String credential = Credentials.basic(proxy.getUsername(), proxy.getPassword());
+                        return response.request().newBuilder()
+                                .header("Proxy-Authorization", credential)
+                                .build();
+                    }
+                };
+                okHttpClientBuilder.proxyAuthenticator(proxyAuthenticator);
+            }
+        }
+
+        OkHttpClient client = okHttpClientBuilder
+                .connectTimeout(30, TimeUnit.SECONDS)
+                .readTimeout(30, TimeUnit.SECONDS)
+                .writeTimeout(30, TimeUnit.SECONDS)
+                .build();
+
         return client;
     }
 
+    /**
+     * Only accepts http or https protocol.
+     *
+     * @param proxy
+     * @return java.net.Proxy.Type
+     */
+    private java.net.Proxy.Type getProxyType(String protocol) {
+        if ("http".equals(protocol) || "https".equals(protocol) || protocol == null) {
+            return java.net.Proxy.Type.HTTP;
+        }
+        return null;
```

Comment:
Perhaps you should return `Proxy.Type.DIRECT` here rather than `null`.

Reply:
Thanks @barryadk . The comment is addressed.